src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -278,10 +280,151 @@ static ss::future<std::vector<resp_resource_t>> alter_topic_configuration(
+                  resource.resource_name));
+          });
+
+        co_return responses;
+    }
+
+    for (const auto& resource : resources) {
+        cluster::config_update_request req;
+
+        if (!resource.resource_name.empty()) {
+            responses.push_back(
+              make_error_alter_config_resource_response<resp_resource_t>(
+                resource,
+                error_code::invalid_config,
+                "Setting broker properties on named brokers is unsupported"));
+            continue;
+        }
+
+        bool errored = false;
+        for (const auto& c : resource.configs) {
+            auto mapped_name = map_config_name(c.name);
+            auto op = static_cast<config_resource_operation>(
+              c.config_operation);
+            if (op == config_resource_operation::set) {
+                req.upsert.push_back(
+                  {ss::sstring(mapped_name), c.value.value()});
+            } else if (op == config_resource_operation::remove) {
+                req.remove.push_back(ss::sstring(mapped_name));
+            } else {
+                responses.push_back(
+                  make_error_alter_config_resource_response<resp_resource_t>(
+                    resource,
+                    error_code::invalid_config,
+                    fmt::format(
+                      "operation {} on broker properties isn't currently "
+                      "supported",
+                      op)));
+                errored = true;
+                continue;
+            }
+        }
+        if (errored) {
+            continue;
+        }
+
+        // Validate contents of the request
+        config::configuration cfg;
+        for (const auto& i : req.upsert) {
+            // Decode to a YAML object because that's what the property
+            // interface expects.
+            // Don't both catching ParserException: this was encoded
+            // just a few lines above.
+            const auto& yaml_value = i.second;
+            auto val = YAML::Load(yaml_value);
+
+            if (!cfg.contains(i.first)) {
+                responses.push_back(
+                  make_error_alter_config_resource_response<resp_resource_t>(
+                    resource,
+                    error_code::invalid_config,
+                    fmt::format("Unknown property '{}'", i.first)));
+                errored = true;
+                continue;
+            }
+            auto& property = cfg.get(i.first);
+            try {
+                property.set_value(val);
+            } catch (...) {
+                responses.push_back(
+                  make_error_alter_config_resource_response<resp_resource_t>(
+                    resource,
+                    error_code::invalid_config,
+                    fmt::format(
+                      "bad property value for '{}': '{}'", i.first, i.second)));
+                errored = true;
+                continue;
```

Comment:
looks like unnecessary continue which ends up reading a little ambiguous being inside a nested for loop